### PR TITLE
Gamepad changes to add custom bindings

### PR DESCRIPTION
**Pull request by wselken**
Added Some minor gamepad changes.

### CODE REVIEW

**Comment by larsiusprime** [comment]
Um, wow. This is pretty cool.

I think I roughly get what this does, but could you explain it a bit? I maintain HaxeFlixel's gamepad API and so I'd like to know exactly what this adds to the equation on the lime level.

File: project/src/backend/sdl/SDLApplication.cpp [hunk 1]
```diff
@@ -131,7 +234,8 @@ namespace lime {
 			case SDL_CONTROLLERBUTTONUP:
 			case SDL_CONTROLLERDEVICEADDED:
 			case SDL_CONTROLLERDEVICEREMOVED:
-				
+			case SDL_JOYDEVICEADDED:
+				printf("game Event %i\n" ,event->type);
```
Comment (Gama11) [inline project/src/backend/sdl/SDLApplication.cpp]: Leftover from debugging?

**Comment by jgranick** [comment]
Could you guys try the latest Lime (after this commit: https://github.com/openfl/lime/commit/5c5430ea0b24a422efd3b91b6164b1fb11d85e91) and see if it helps?

I'm wondering if the JOYSTICK_ADDED is needed. Is it going to add a regular joystick as if it were a gamepad if we do that? Should we be concerned about that problem?

**Comment by wselken** [comment]
it works with your current code, expect the JOYSTICK_ADDED needs to call ProcessGamePadEvent, 
I'm don't see why this would be a problem, since ProcessGamePadEvent checks to make sure the joystick is gamepad before adding it. if its not a gamepad, it will just do nothing, which is the current use case for it now. (from the looks of the code.)

**Pull request closed without merging**